Puc/v4p2/Plugin/UpdateChecker.php

```diff
@@ -399,6 +400,49 @@ public function addCheckForUpdatesLink($pluginMeta, $pluginFile) {
 		}
 
 		/**
+		 * Add a "View Details" link to the plugin row in the "Plugins" page. By default,
+		 * the new link will appear after the "Visit plugin site" link.
+		 *
+		 * You can change the link text by using the "puc_view_details_link-$slug" filter.
+		 * Returning an empty string from the filter will disable the link.
+		 *
+		 * @param array $pluginMeta Array of meta links.
+		 * @param string $pluginFile
+		 * @return array
+		 */
+		public function addViewDetailsLink($pluginMeta, $pluginFile) {
+			$isRelevant = ($pluginFile == $this->pluginFile)
+				|| (!empty($this->muPluginFile) && $pluginFile == $this->muPluginFile);
+
+			if ( $isRelevant && $this->userCanInstallUpdates() ) {
+				//Check if view-details is already among the links (because there is an update)
+				$viewDetailsExists = false;
+				foreach ($pluginMeta as $existingMeta) {
+					if (strpos($existingMeta, 'tab=plugin-information') !== false) {
+						$viewDetailsExists = true;
+					}
+				}
+				$linkText = apply_filters(
+					$this->getUniqueName('view_details_link'),
+					__( 'View details' )
+				);
+				if ( !empty($linkText) && !$viewDetailsExists ) {
+					$pluginTitle = $this->getPluginTitle();
```

Comment:
`getPluginTitle()` could be slow since it has to hit the disk. Consider using `$pluginData['Name']` instead.